src/ansys/units/unit.py

```diff
@@ -7,19 +8,42 @@ class Unit:
 
     Parameters
     ----------
-    Name: str,
-        Name of the unit or string chain of combined units.
-    Config: dict
-        dictionary of unit properties.
+    units: str, optional
+        Name of the unit or string chain of combined units
+    config: dict, optional
+        dictionary of unit properties
+    dimensions: Dimensions, optional
+        An instance of the Dimensions class.
+    unit_sys: str, optional
+        Define the unit system for base units of dimension,
+        default is SI.
 
     Returns
     -------
     Unit
-        Unit instance
+        Unit instance.
     """
 
-    def __init__(self, units: str, config: dict = None):
-        self._name = units
+    def __init__(
+        self,
+        units: str = None,
+        config: dict = None,
+        dimensions: ansunits.Dimensions = None,
+        unit_sys: ansunits.UnitSystem = None,
+    ):
+        if units and dimensions:
```

Comment:
Maybe we can update the logic here. Instead of saying that you can't pass both, we can validate that dims == units.dims.